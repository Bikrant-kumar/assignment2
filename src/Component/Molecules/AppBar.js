import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import RestaurantIcon from '@material-ui/icons/Restaurant';
import { grey } from '@material-ui/core/colors';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import { Link } from 'react-router-dom';
import Label from '../Atoms/Label';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    background: 'linear-gradient(45deg, #21CBF3 30%, #2196F3 90%)',
    boxShadow: '0 3px 5px 2px rgba(33, 203, 243, .3)',
  },
  menuButton: {
    marginRight: theme.spacing(1),
  },
  displayRight:{
    marginLeft:'32%',
  },
}));

export default function DenseAppBar() {
  const classes = useStyles();
  const [value, setValue] = React.useState(0);
  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  return (
      <AppBar position="static" className={classes.root}>
        <Toolbar variant="dense">
          <IconButton edge="start" className={classes.menuButton} aria-label="menu">
            <RestaurantIcon />
          </IconButton>
          <Typography variant="h5" style={{ color: grey[900] }} >
            <Label text={"MY Restaurant"}/> 
          </Typography>
          <Tabs value={value}
              onChange={handleChange}
              textColor="secondary"
              className={classes.displayRight}
              >
                <Tab label="MANAGER" component={Link} to="/"/> 
                <Tab label="MENU" component={Link} to="/Menu"/>
                <Tab label="ORDER" component={Link} to="/Order"/> 
                <Tab label="REVIEW" component={Link} to="/Review"/> 
            </Tabs>
            </Toolbar>
      </AppBar>
  );
}