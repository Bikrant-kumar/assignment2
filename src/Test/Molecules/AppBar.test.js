import React from 'react';
import { screen,render } from '@testing-library/react';
import AppBar from '../../Component/Molecules/AppBar';
import renderer from 'react-test-renderer';

describe(AppBar ,() => {

   test('AppBar Render',() => {
      render(<AppBar /> );
      expect(screen.getByText(/MY Restaurant/)).toBeInTheDocument();
  });

   test('test AppBar with snapshot',()=>{
      const component= renderer.create(<AppBar/>).toJSON();
      expect(component).toMatchSnapshot();
   });

})


