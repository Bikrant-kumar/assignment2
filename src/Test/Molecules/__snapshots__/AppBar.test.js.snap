// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DenseAppBar test AppBar with snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-5 MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
  >
    <button
      aria-label="menu"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-6 MuiIconButton-edgeStart"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 9H9V2H7v7H5V2H3v7c0 2.12 1.66 3.84 3.75 3.97V22h2.5v-9.03C11.34 12.84 13 11.12 13 9V2h-2v7zm5-3v8h2.5v8H21V2c-2.76 0-5 2.24-5 4z"
          />
        </svg>
      </span>
    </button>
    <h5
      className="MuiTypography-root MuiTypography-h5"
      style={
        Object {
          "color": "#212121",
        }
      }
    >
      <div
        className="MuiBox-root MuiBox-root-8"
      >
        MY Restaurant
      </div>
    </h5>
    <div
      className="makeStyles-displayRight-7"
    />
  </div>
</header>
`;
